### ISSUE TITLE
Clicking a problem opens new absolute path window.

### ISSUE THREAD

**Issue by jalanh11640** [microsoft/vscode-cpptools#9871]
### Environment

- OS and version: macOS Monterey (12.5.1)
- VS Code: 1.68.1
- C/C++ extension: v1.12.4 (Extension is enabled on 'SSH:myhost')
- OS and version of remote machine (if applicable): Red Hat Enterprise Linux Workstation release 7.7 (Maipo)

### Bug Summary and Steps to Reproduce

Bug Summary: Clicking on a problem in the 'problem pane' opens the file in the editor with an absolute path even if file (with a relative path to the workspace) is already open.

Steps to reproduce:
1. Create x/x.exe/x.cpp with 'x' in the file.
2. Create x/x.code-workspace with the following..
        {
	        "folders": [
		        {
			        "path": "."
		        }
	        ]
        }

3. Two problems will come up in the problem pane.
     this declaration has no storage class or type specifier C/C++(77) [Ln1, Col 1]
     expected a ';' C/C++(54) [Ln 1, Col 1]

    Click on either problem.
4. New file opens. Breadcrumbs view shows full path.

### Other Extensions

Remote-SSH: v0.84.0 (Locally installed)
Remote - SSH: Editing Configuration Files: v0.80.0 (Locally Installed

### Additional Information

I appologize of this is the wrong place for this bug report.

**Comment by Colengms** [comment]
Hi @jalanh11640 .  Could you provide more information about this scenario and your repro, such as the paths involved and their differences?  The C/C++ Extension should be providing IntelliSense (and 'problems' associated with `C/C++`), for files that are already open in the editor.  Those problems should be implicitly associated with paths to files that are already open.  I can't repro simply, and I don't believe we've had reports about this being a general issue, so I suspect there are details about your environment or scenario that we'll need to better understand to investigate further.

**Comment by jalanh11640** [comment]
Hey @Colengms.  While collecting data to provide answers to you question, I think I've figured out the issue.

When I opened up the x.code-workspace file I navigated through a symbolic link (/home/jeff/projects/x/x.code-workspace, where 'projects' -> '/project_store/projects').  When I open the file by clicking on the file in the 'file navigator' the breadcrumb view shows "x > x.exe > x.cpp".  When I click on one of the problems, the breadcrumb view for the file shows "project_store > projects > x > x.exe > x.cpp".

If I instead open the workspace via the 'real' path (/project_store/projects/x/x.code-workspace) then I don't experience the issue.
